### PR TITLE
Introduce submit button as a fieldtype for form

### PR DESCRIPTION
**Pull request by kennyt**

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: docs/src/Form/index.js [hunk 1]
```diff
@@ -42,8 +42,9 @@ class FormExample extends React.Component {
         showLabel: 'Accept terms of service',
         required: true,
         label: 'Single checkbox',
+        value: true,
         validation: function (value) {
-          return value && value.checked;
+          return value;
```
Comment (mlunoe) [inline docs/src/Form/index.js]: Why did this change?

File: src/Form/FieldCheckboxMultiple.js [hunk 2]
```diff
@@ -18,8 +18,7 @@ class FieldCheckboxMultiple extends Util.mixin(BindMixin) {
       props.name,
       {
         name: name,
-        checked: modelState.checked,
-        indeterminate: modelState.indeterminate
+        checked: modelState
```
Comment (mlunoe) [inline src/Form/FieldCheckboxMultiple.js]: Why did this change?

File: docs/src/Form/index.js (hunk 1 above)
Comment (kennyt) [inline docs/src/Form/index.js]: because value is now whether it's checked or not

File: src/Form/FieldCheckboxMultiple.js (hunk 2 above)
Comment (kennyt) [inline src/Form/FieldCheckboxMultiple.js]: because modelState is now the checked value

Comment (jfurrow) [inline src/Form/FieldCheckboxMultiple.js]: So `checked` can be one of `true`, `false`, or `'indeterminate'`?

Comment (kennyt) [inline src/Form/FieldCheckboxMultiple.js]: nope, indeterminate is still a property on the definition. it's just that indeterminate will come from the parent, the checkbox will never hold the value of indeterminate in state

Comment (jfurrow) [inline src/Form/FieldCheckboxMultiple.js]: ahh cool!

**Comment by jfurrow** [comment]
:shipit:

**Pull request merged**